gravitee-apim-console-webui/src/management/api/runtime-logs-v4/runtime-logs-messages/api-runtime-logs-messages.component.ts

```diff
@@ -0,0 +1,97 @@
+import { ConnectorPlugin, ConnectorType, MessageLog } from '../../../../entities/management-api-v2';
+import { IconService } from '../../../../services-ngx/icon.service';
+import { ConnectorPluginsV2Service } from '../../../../services-ngx/connector-plugins-v2.service';
+
+@Component({
+  selector: 'api-runtime-logs-messages',
+  template: require('./api-runtime-logs-messages.component.html'),
+  styles: [require('./api-runtime-logs-messages.component.scss')],
+})
+export class ApiRuntimeLogsMessagesComponent implements OnInit {
+  private readonly pageSize: number = 5;
+  public connectorIcons: { [key: string]: string } = {};
+  public messageLogs$: BehaviorSubject<MessageLog[]> = new BehaviorSubject<MessageLog[]>([]);
+  public pageIndex = 1;
+  public pageCount: number;
+
+  constructor(
+    @Inject(UIRouterStateParams) private readonly ajsStateParams: StateParams,
+    private readonly apiLogsService: ApiLogsV2Service,
+    private readonly connectorPluginsV2Service: ConnectorPluginsV2Service,
+    private readonly iconService: IconService,
+  ) {}
+
+  public ngOnInit(): void {
+    this.loadMessages(this.pageIndex);
+  }
+
+  private loadMessages(pageIndex: number): void {
+    this.apiLogsService
+      .searchMessageLogs(this.ajsStateParams.apiId, this.ajsStateParams.requestId, pageIndex, this.pageSize)
+      .pipe(
+        map((messageLogs) => {
+          this.messageLogs$.next([...this.messageLogs$.getValue(), ...messageLogs.data]);
+          this.pageIndex += 1;
+          this.pageCount = messageLogs.pagination.pageCount;
+          return messageLogs.data;
+        }),
+        switchMap((messageLogs: MessageLog[]) =>
+          uniqBy(
+            messageLogs.map((messageLog) => ({
+              connectorId: messageLog.connectorId,
+              connectorType: messageLog.connectorType,
+            })),
+            'connectorId',
+          ),
+        ),
+        tap((entry: { connectorId: string; connectorType: ConnectorType }) => {
+          if (!this.connectorIcons[entry.connectorId]) {
+            this.loadConnectorIcon(entry.connectorType, entry.connectorId);
+          }
+        }),
+      )
+      .subscribe();
+  }
+
+  public loadMoreMessages(): void {
+    this.loadMessages(this.pageIndex);
+  }
+
+  private loadConnectorIcon(connectorType: ConnectorType, connectorId: string) {
+    if (connectorType === 'ENDPOINT') {
+      this.connectorPluginsV2Service.getEndpointPlugin(connectorId).subscribe((connectorPlugin: ConnectorPlugin) => {
+        this.connectorIcons[connectorId] = this.iconService.registerSvg(connectorId, connectorPlugin.icon);
+      });
+    }
+
+    if (connectorType === 'ENTRYPOINT') {
+      this.connectorPluginsV2Service.getEntrypointPlugin(connectorId).subscribe((connectorPlugin: ConnectorPlugin) => {
+        this.connectorIcons[connectorId] = this.iconService.registerSvg(connectorId, connectorPlugin.icon);
+      });
```

Comment:
would it worth to load all the icons of connectors in local storage ?

Reply:
I wanted to avoid to load all connectors icons to be honnest. I suppose that the customers won't have APIs with all the entrypoints and endpoint types in one API no ? I can load everything if we find it usefull but has it's possible to have its individuality... 🤷🏼